utils/scripts/find_duplicate_read_sets/find_duplicate_read_sets.py

```diff
@@ -0,0 +1,118 @@
+import boto3
+import pandas as pd
+import argparse
+
+
+def find_duplicates(seq_store_ids, omics_client, output_path):
+    """
+    Iterates through all the read sets in all the sequence stores and returns a dataframe with
+    the read sets that share an ETag. This will find duplicates for read sets across sequence 
+    stores.
+
+    """
+
+    # intiate a blank dataframe to store the information for read sets
+    df = pd.DataFrame(columns=['sequence_store_id','read_set_id','name','type','source', 'etag'])
+
+    # setup the paginator to use list read sets
+    paginator = omics_client.get_paginator('list_read_sets')
+
+    # iterate by sequence store.  You can modify this loop if you want to limit to finding duplicates just within the sequence store 
+    for seq_store in seq_store_ids:
+
+        # iterate by page within a read set
+        for page in paginator.paginate(sequenceStoreId=seq_store):
+
+            # get just the read set information
+            for rs in page['readSets']:
+                read_set_id = rs['id']
+                if rs.get('etag',''):
+
+                    # there may be 2 sources if the file type is a FASTQ
+                    etag_source_1 = rs['etag'].get('source1','')
+                    etag_source_2 = rs['etag'].get('source2','')
+                    file_type = rs.get('fileType')
+                    name = rs.get('name')
+
+                    # only log if a etag was found
+                    if etag_source_1:
+                        s1 = {
+                            'sequence_store_id': seq_store,
+                            'read_set_id': read_set_id, 
+                            'name': name,
+                            'type': file_type,
+                            'source': 'source 1', 
+                            'etag': etag_source_1
+                        } 
+                        df = pd.concat([df, pd.DataFrame([s1])])
+                    if etag_source_2:
+                        s2 = {
+                            'sequence_store_id': seq_store,
+                            'read_set_id': read_set_id, 
+                            'name': name,
+                            'type': file_type,
+                            'source': 'source 2', 
+                            'etag': etag_source_2
+                        } 
+                        df = pd.concat([df, pd.DataFrame([s2])])
+
+    # filters to only duplicates based on the etag field
+    filtered_df = df[df.duplicated(subset='etag', keep=False)]
```

Comment:
Is `pandas` really needed here? Can you achieve the same results with lighter weight packages in the stdlib - e.g. a combination of `csv` and `set` operations?

Reply:
based on discussions, switching to using stdlib could lead to performance concerns.